activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -117,7 +123,8 @@ def indexes(table_name) # :nodoc:
             if indkey.include?(0)
               columns = expressions
             else
-              columns = column_names_from_column_numbers(oid, indkey)
+              decoder = PG::TextDecoder::Array.new
```

Comment:
Hi @fatkodima 


I have this index in my `structure.sql`
```sql
ALTER TABLE ONLY public.sidestore_app_images
    ADD CONSTRAINT uniq_rails_fd6debf15f UNIQUE (sidestore_app_id, "position") DEFERRABLE INITIALLY DEFERRED;
```

As you can see the `position` column is surrounded by quotes.

![image](https://github.com/user-attachments/assets/f344138e-9b19-4c53-9a17-37fad6f32053)

Unfortunately as you can see the `decode` method returns an array of columns but with the `position` column surrounded by quotes

I have the same issue with the `order` column that is also surrounded by quotes with postgresql.
```sql
CREATE UNIQUE INDEX index_documents_on_order_and_ar24_registered_base_id ON public.documents USING btree (registered_base_id, "order");
```

Do you think we should further sanitize here so that the array removes any extra `""` or it should be the application/gem ( here `database_consistency` ) to cleanup the columns array for these special columns ( `position` and `order` )

Thanks!

Reply:
Should be as was before. Before it stripped extra quotes. I will push a fix soon.

Reply:
Thanks

Reply:
Checked, rails returns correct values for the `column`:
```ruby
[#<struct ActiveRecord::ConnectionAdapters::PostgreSQL::UniqueConstraintDefinition
  table_name=:messages,
  column=["sidestore_app_id", "position"],
  options={:name=>"uniq_rails_5b52e33238", :nulls_not_distinct=>false, :deferrable=>:deferred}>]
```

Why do you manually call `PG:TextDecoder....` on it?

Reply:
Sorry I just saw your message. I don't manually call TextDecoder on it, i put a `binding.irb` right inside the rails source code ( line 125 `activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb` )

And I manually called `PG:TextDecoder` because you did it on line 126. hence the screenshot @fatkodima

Reply:
If the problem persists, can you create a reproduction test script using https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_record_migrations.rb template?

Reply:
I mean I did the binding.irb on your branch, but it's the same on the main branch since it calls the same decoder
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/02db9143-81bc-4fe7-b4a0-561d41ff328d" />

and 
<img width="476" alt="image" src="https://github.com/user-attachments/assets/c715adc2-7d5e-49c4-9120-9be2ba4dbf05" />


I'll try to create a reproduction script ( hopefully i'll succeed), it's just that the database on `row[6]` returns an unparsed string with those additional quotes.

Reply:
Ok I managed to replicate. I opened an issue : https://github.com/rails/rails/issues/54057 @fatkodima